internal/generate/olm-catalog/package_manifest.go

```diff
@@ -96,41 +95,81 @@ func (g pkgGenerator) Generate() error {
 // generate either reads an existing package manifest or creates a new
 // manifest and modifies it based on values set in s.
 func (g pkgGenerator) generate() (map[string][]byte, error) {
+	fileName, pkg, err := g.buildPackageManifest()
+	if err != nil {
+		return nil, err
+	}
+
+	g.setChannels(&pkg)
+	g.sortChannelsByName(&pkg)
+
+	if err := validatePackageManifest(&pkg); err != nil {
+		return nil, fmt.Errorf("error validating package manifest %s: %v", pkg.PackageName, err)
+	}
+
+	b, err := yaml.Marshal(pkg)
+	if err != nil {
+		return nil, err
+	}
+	fileMap := map[string][]byte{
+		fileName: b,
+	}
+	return fileMap, nil
+}
+
+// buildPackageManifest will build the registry.PackageManifest which is the structure used by OLM
+// by reading the files found in the path.
+func (g pkgGenerator) buildPackageManifest() (string, registry.PackageManifest, error) {
+	pkg := registry.PackageManifest{}
 	fileName := getPkgFileName(g.OperatorName)
 	path := filepath.Join(g.Inputs[ManifestsDirKey], fileName)
-	pkg := registry.PackageManifest{}
 	if _, err := os.Stat(path); err == nil {
 		b, err := ioutil.ReadFile(path)
 		if err != nil {
-			return nil, fmt.Errorf("failed to read package manifest %s: %v", path, err)
+			return "", pkg, fmt.Errorf("failed to read package manifest %s: %v", path, err)
 		}
 		if err = yaml.Unmarshal(b, &pkg); err != nil {
-			return nil, fmt.Errorf("failed to unmarshal package manifest %s: %v", path, err)
-		}
-		if err := registryutil.ValidatePackageManifest(&pkg); err != nil {
-			return nil, fmt.Errorf("error validating package manifest %s: %v", pkg.PackageName, err)
+			return "", pkg, fmt.Errorf("failed to unmarshal package manifest %s: %v", path, err)
 		}
 	} else if os.IsNotExist(err) {
 		pkg = newPackageManifest(g.OperatorName, g.channel, g.csvVersion)
 	} else {
-		return nil, fmt.Errorf("error reading package manifest %s: %v", path, err)
+		return "", pkg, fmt.Errorf("error reading package manifest %s: %v", path, err)
 	}
+	return fileName, pkg, nil
+}
 
-	g.setChannels(&pkg)
+// sortChannelsByName sort the channels by the name
+// NOTE: sorting makes the channel order always consistent when appending new channels
+func (g pkgGenerator) sortChannelsByName(pkg *registry.PackageManifest) {
```

Comment:
This doesn't need to be a method.

Reply:
Note that:
- Separating our code in multiple methods allows us to reuse them. It also makes our code a lot more organized.
- More methods will not slow our project down to any noticeable degree. 
- Smaller methods are easier to read and ingest. They also are more likely to have only one duty. Both very good things.
- Many methods can force a reader to maintain a large "mental stack" as they read through code and have to keep track of a lot of method calls. As long as methods do only one thing (single responsibility/cohesion), and are named well, this is usually not a problem.

The two principles we should be keeping in mind are:
- DRY - Don't Repeat Yourself
- KISS - Keep It Simple Stupid.

See that it is easier and faster understand the code when we see:

```go
	g.setChannels(&pkg)
	g.sortChannelsByName(&pkg)
```

Compared with: 

```go
	g.setChannels(&pkg)
	sort.Slice(pkg.Channels, func(i int, j int) bool {
		return pkg.Channels[i].Name < pkg.Channels[j].Name
	})
```

**Observation**

Sorry, but I do not see any reason/justification to revert this change or why not follow with since it is a good practice in any language. It has not relation with [Premature Optimization](https://en.wikipedia.org/wiki/Program_optimization)

> "Premature optimization" is a phrase used to describe a situation where a programmer lets performance considerations affect the design of a piece of code. This can result in a design that is not as clean as it could have been or code that is incorrect, because the code is complicated by the optimization and the programmer is distracted by optimizing.

Reply:
I didn't say it can't be a function. No fields of `pkgGenerator` are used in sortChannelsByName, so it shouldn't be a method on `g`.

Reply:
Great spot done 👍